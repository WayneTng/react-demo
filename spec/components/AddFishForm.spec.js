import React from 'react'
import renderer from 'react-test-renderer'
import fishStoreFactory from 'stores/Fish'
import AddFishForm from 'components/AddFishForm'

describe('<AddFishForm />', () => {
  it('renders a fish form', () => {
    const store = fishStoreFactory.getStore('testStore')
    const addFishForm = renderer.create(
      <AddFishForm addFish={store.addFish} />
    )

    const tree = addFishForm.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
