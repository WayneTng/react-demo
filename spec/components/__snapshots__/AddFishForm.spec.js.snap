exports[`<AddFIshForm /> renders a fish form 1`] = `
<form
  className="fish-edit"
  onSubmit={[Function onSubmit]}>
  <input
    placeholder="Fish Name"
    type="text" />
  <input
    placeholder="Fish Price"
    type="text" />
  <select>
    <option
      value="available">
       Fresh! 
    </option>
    <option
      value="unavailable">
       Sold Out! 
    </option>
  </select>
  <textarea
    placeholder="Fish Desc" />
  <input
    placeholder="Fish Image"
    type="text" />
  <button
    type="submit">
     + Add Item 
  </button>
</form>
`;

exports[`<AddFishForm /> renders a fish form 1`] = `
<form
  className="fish-edit"
  onSubmit={[Function onSubmit]}>
  <input
    placeholder="Fish Name"
    type="text" />
  <input
    placeholder="Fish Price"
    type="text" />
  <select>
    <option
      value="available">
       Fresh! 
    </option>
    <option
      value="unavailable">
       Sold Out! 
    </option>
  </select>
  <textarea
    placeholder="Fish Desc" />
  <input
    placeholder="Fish Image"
    type="text" />
  <button
    type="submit">
     + Add Item 
  </button>
</form>
`;
